human-forecasts/server-distribution-code.R

```diff
@@ -0,0 +1,432 @@
+               }
+             })
+observeEvent(c(input$propagate_3), 
+             {
+               for (i in 4:4) {
+                 rv$median_latent[i] <- rv$median_latent[3]
+                 rv$width_latent[i] <- rv$width_latent[3] #+ 0.01 * (i - 3)
+                 updateNumericInput(session,
+                                    paste0("median_forecast_", i),
+                                    value = round(rv$median_latent[i], 0))
+                 
+                 updateNumericInput(session,
+                                    paste0("width_", i),
+                                    value = round(rv$width_latent[i], 2))
+               }
+             })
+
+# update
+observeEvent(c(input$update_1, input$update_2, input$update_3, input$update_4), 
+             {
+               update_values()
+             })
+
+
+# change values by numeric input
+observeEvent(input$median_forecast_1,
+             {
+               rv$median_latent[1] <- input$median_forecast_1
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$median_forecast_2,
+             {
+               rv$median_latent[2] <- input$median_forecast_2
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$median_forecast_3,
+             {
+               rv$median_latent[3] <- input$median_forecast_3
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$median_forecast_4,
+             {
+               rv$median_latent[4] <- input$median_forecast_4
+               # update_values()
+             }, 
+             priority = 99)
+
+
+observeEvent(input$width_1,
+             {
+               rv$width_latent[1] <- input$width_1
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$width_2,
+             {
+               rv$width_latent[2] <- input$width_2
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$width_3,
+             {
+               rv$width_latent[3] <- input$width_3
+               # update_values()
+             }, 
+             priority = 99)
+observeEvent(input$width_4,
+             {
+               rv$width_latent[4] <- input$width_4
+               # update_values()
+             }, 
+             priority = 99)
+
+
+
+
+
+
+observeEvent(input$tooltip,
+             {
+               if (input$tooltip == "yes") {
+                 addTooltip(session = session, 
+                            id = "tooltip", 
+                            title = "Toggle tooltips on and off", 
+                            placement = "bottom", trigger = "hover",
+                            options = NULL)
+                 
+                 addTooltip(session = session, 
+                            id = "tooltip", 
+                            title = "Toggle tooltips on and off", 
```

Comment:
Helper fn to reduce code replication

Reply:
e.g. something where I would pass in a list of pairs corresponding to "id" and "tooltip text"?